streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamPrintTest.java

```diff
@@ -82,9 +83,42 @@ public void testPrintKeyValueWithName() {
             driver.process(topicName, record.key, record.value);
         }
         printWriter.flush();
-        final String[] flushOutDatas = new String(byteOutStream.toByteArray(), Charset.forName("UTF-8")).split("\n");
+        final String[] flushOutDatas = new String(byteOutStream.toByteArray(), Charset.forName("UTF-8")).split("\\r*\\n");
         for (int i = 0; i < flushOutDatas.length; i++) {
-            assertEquals(flushOutDatas[i], expectedResult[i]);
+            assertEquals(expectedResult[i], flushOutDatas[i]);
+        }
+    }
+
+    @Test
+    public void testPrintKeyValueMapperWithName() {
+        final KeyValueMapper<Integer, String, String> mapper = new KeyValueMapper<Integer, String, String>() {
+            @Override
+            public String apply(Integer key, String value) {
+                return String.format("(%d, %s)", key, value);
+            }
+        };
+        final KStreamPrint<Integer, String> kStreamPrint = new KStreamPrint<>(new PrintForeachAction(printWriter, mapper, "test-stream"), intSerd, stringSerd);
```

Comment:
note that I just do it in local and not yet to PR, sorry for an ambiguous statement